### DISCUSSION TITLE
How hvac_action is calculated if my ubderlying device doesn't expose one ?

### DISCUSSION THREAD

**Discussion by jmcollin78** [jmcollin78/versatile_thermostat#626] [Q&A]
Good question....

For devices that do not have an hvac_action, I simulate one:

For `over_valve`, the valve opening must be > min,
for `over_climate`, it's like you said, the difference between the target and the current temp must be > 0 (or < 0 in Cooling mode).
And be careful with multi-devices Vtherm, at least the underlying must be its hvac_action (simulated or not) so that its hvac_action is ‹ on ›.

For over_switches, at least one of the controlled switches must be 'on'.

So indeed you can see differences compared to the actual ignition state.
The ideal would be to have an `hvac_action` attribute calculated according to the real state of the device (if it has one or if you can simulate one of course).

**Comment by lucvan** [comment]
Would this for some reason cause an AC unit that does not expose HVAC action to enable the central boiler if it was in cooling mode, but in a Vtherm with a TRV also?

**Reply by jmcollin78** [reply]
> but in a Vtherm with a TRV also?

I don't understand what it means ? Can you give me more details on your configuration ?
